libknet/logging.c

```diff
@@ -244,6 +249,19 @@ void log_msg(knet_handle_t knet_h, uint8_t subsystem, uint8_t msglevel,
 	len = strlen(msg.msg);
 	msg.msg[len+1] = '\n';
 
+	/*
+	 * we need to serialize log flush to the fd.
+	 * the kernel can potentially break the write in multiple chunks
+	 * and multiple calling threads could generate unreadable logs.
+	 *
+	 * this needs to be done at global level (and not per-handle)
+	 * as multiple handle could be sharing the same log fd.
+	 */
+	lock_flush = pthread_mutex_lock(&log_flush);
```

Comment:
Are you comfortable introducing a potentially blocking operation into the logging code? I understand that this lock won't be held for long because the logging descriptor is non-blocking, but if something unexpected happens... Moreover, this does not protect from losing log messages because of the non-blocking nature again. On the other hand, POSIX guarantees atomicity of pipe writes smaller than `PIPE_BUF`, which is at least 512 bytes, twice the size of current `struct knet_log_msg`, so we're good as long as we stick to using pipes.

Reply:
I'm with feri on this, having two locks around the logging code seems excessive to me. I know it's possible for two handles to share the same logging fd but it seems excessive punishment for a very rare occurrance and a 'possible' minor logging annoyance.